back/engines/commercial/analytics/spec/models/analytics/dimension_project_spec.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Analytics::DimensionProject, type: :model do
+  it 'can create a project dimension by creating a project' do
+    project = create(:project)
+    project_dimension = described_class.first
+    assert(project_dimension.id == project.id)
+  end
```

Comment:
Not perfect, but to give an idea of how I would write this:
```suggestion
  context 'when a project is created' do
    let!(:project) { create(:project) }
    
    it 'is also available as a project dimension' do 
      described_class.find(project.id)
    end
  end
```

Reply:
or...
```suggestion
  it 'is created for each project' do 
    projects = create_list(:project, 2)
    project_ids = projects.pluck(:id)
    expect(described_class.ids).to match_array(project_ids)
  end
```